### PR TITLE
Update `astro check` to latest version of the language server

### PR DESCRIPTION
**Pull request by Princesseuh**
## Changes

Update the language-server dep to the latest version. This version of the language server does not bundle typescript anymore, as such we need to add it as a dep of Astro itself

## Testing

Tests still pass!

## Docs

N/A

### CODE REVIEW

File: packages/astro/package.json [hunk 1]
```diff
@@ -147,6 +147,7 @@
     "strip-ansi": "^7.0.1",
     "supports-esm": "^1.0.0",
     "tsconfig-resolver": "^3.0.1",
+    "typescript": "*",
```
Comment (matthewp) [inline packages/astro/package.json:150]: is this safe?

Comment (Princesseuh) [reply packages/astro/package.json:150]: *yes and no*, if you don't have TypeScript installed in your project, then updating Astro might update the TypeScript version which might break the type checking if TypeScript introduce a breaking change

If you do have TypeScript installed in your project, it should normally just use the workspace's version and so it won't break

Comment (matthewp) [reply packages/astro/package.json:150]: Curious why this was done in this PR. Can we not update the language server without this?

Comment (Princesseuh) [reply packages/astro/package.json:150]: Starting from version 0.24.0, the language server doesn't bundle TypeScript anymore (to allow users to control the version, like they would for .ts files and other reasons). 

So yeah, can't upgrade the language server without breaking `astro check`, so we bundle TypeScript into Astro itself instead. In the next update where we can do breaking changes we'll remove this and instead offer a message that says to install TypeScript to use `astro check`

**Comment by matthewp** [comment]
Since this is a big change let's release it when the next minor goes out.

**Pull request merged**